### PR TITLE
Develop Dashboard Page UI

### PR DESCRIPTION
**Pull request by sdidulosence1**
---
[Develop Dashboard Page UI](https://www.notion.so/PRO-MVP-Develop-Dashboard-Page-UI-8fbb642e9a094d9d84a2eb248e9b0202?pvs=4)
---

---

#### Feature Description
- Design and implement a Dashboard Page UI for providers to manage scholarships.
- Include a table displaying all scholarships with columns for Scholarship name, start date, end date, status, and actions (view, edit, delete).
- Implement sorting functionality for each column in the table, allowing providers to sort the data in ascending or descending order by clicking on the column headers.
- Include buttons for adding scholarships and adding scholarships via CSV file upload.

---

#### Steps to Reproduce the Feature

Page is located in: http://localhost:3000/provider/providerDashboard

---

#### Expected Behavior

N/A

---

#### Test Environment

- Browser: [e.g. Chrome, Safarri, Firefox]
- Test Scenarios:
  - [ ] Did the Test Feature Pass?
  - [ ] Did you check the features on different browsers?

---

#### Additional Details / Context

> Add any other context about the problem here.

### CODE REVIEW

File: frontend/src/components/DashboardTable/DashboardTable.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        rows={rows}
+        columns={columns}
+        initialState={{
+          pagination: {
+            paginationModel: { page: 0, pageSize: 5 },
+          },
+        }}
+        pageSizeOptions={[5, 10]}
+        checkboxSelection
+        sx={{
```
Comment (sence1roandayne) [inline frontend/src/components/DashboardTable/DashboardTable.tsx:171]: Add these to change colors

```
     '& .MuiDataGrid-columnHeaders': {
            backgroundColor: '#AFC3D9',
     },
     '& .MuiDataGrid-footerContainer': {
            backgroundColor: '#AFC3D9', // Change table header color
     },
     '& .MuiDataGrid-row': {
            '&:nth-of-type(odd)': {
              backgroundColor: '#D8D8D8', // Change background color of odd rows
            },
            '&:nth-of-type(even)': {
              backgroundColor: '#F1F1F1', // Change background color of odd rows
            },
     },
```

File: frontend/src/components/DashboardTable/DashboardTable.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { DataGrid, GridActionsCellItem } from '@mui/x-data-grid';
+import { Button } from '@mui/material';
+
+const columns = [
+  { field: 'id', headerName: 'ID', width: 70 },
+  { field: 'scholarshipName', headerName: 'Scholarship Name', width: 130 },
+  { field: 'startDate', headerName: 'Start Date', type: "date", width: 130 },
+  { field: 'endDate', headerName: 'End Date', type: "date", width: 130 },
+  { field: 'status', headerName: 'Status', width: 130 },
+  { field: 'actions', headerName: 'Actions', type: "actions", width: 130,
```
Comment (sence1roandayne) [inline frontend/src/components/DashboardTable/DashboardTable.tsx]: Let's make the width bigger then adjust the width of the whole table

**Review by sence1roandayne — CHANGES_REQUESTED** [review]

**Comment by sdidulosence1** [comment]
@sence1roandayne 
Hi Roan-san, I think okay na ung page, mejo mahirap lang i-tweak pa ung mobile version (baka need gawin na dropdown ung buttons sa action if mobile). Fix ko nalang cguro ito later on. 🙇‍♀️

File: frontend/src/components/DashboardTable/DashboardTable.tsx (hunk 1 above)
Comment (sdidulosence1) [reply frontend/src/components/DashboardTable/DashboardTable.tsx:171]: Thank you, I added this 🙏

File: frontend/src/components/DashboardTable/DashboardTable.tsx (hunk 2 above)
Comment (sdidulosence1) [reply frontend/src/components/DashboardTable/DashboardTable.tsx]: Thank you, I have adjusted this part 🙏

**Review by sence1roandayne — APPROVED** [review]
Added the integration here.

LGTM

**Pull request merged**